src/frontend/js/widgets/Dashboard/components/DashboardItem/Order/OrderStateMessage/index.tsx

```diff
@@ -0,0 +1,70 @@
+import { FormattedMessage } from 'react-intl';
+
+import { Order, OrderState } from 'types/Joanie';
+import { StringHelper } from 'utils/StringHelper';
+
+export const messages = {
+  statusWaitingSignature: {
+    id: 'components.DashboardItemOrder.statusWaitingSignature',
+    description:
+      "Status shown on the dashboard order item when order is validated with contract's learner signature missing.",
+    defaultMessage: 'Signature required',
+  },
+  statusWaitingCountersignature: {
+    id: 'components.DashboardItemOrder.statusWaitingCountersignature',
+    description:
+      "Status shown on the dashboard order item when order is validated with contract's organization signature missing.",
+    defaultMessage: 'Awaiting organization signature',
+  },
+  statusOnGoing: {
+    id: 'components.DashboardItemOrder.statusOnGoing',
+    description:
+      'Status shown on the dashboard order item when order is validated with no certificate',
+    defaultMessage: 'On going',
+  },
+  statusCompleted: {
+    id: 'components.DashboardItemOrder.statusCompleted',
+    description:
+      'Status shown on the dashboard order item when order is validated with certificate',
+    defaultMessage: 'Completed',
+  },
+  statusOther: {
+    id: 'components.DashboardItemOrder.statusOther',
+    description: 'Status shown on the dashboard order item when order is not validated',
+    defaultMessage: '{state}',
+  },
+};
+
+interface OrderStateMessageProps {
+  order: Order;
+}
+
+const OrderStateMessage = ({ order }: OrderStateMessageProps) => {
+  const { contract, certificate } = order;
+
+  if (contract) {
+    if (contract.nb_signatures === 0) {
+      return <FormattedMessage {...messages.statusWaitingSignature} />;
+    }
+    if (contract.nb_signatures < 2) {
```

Comment:
I think it could be a better idea to define a constant instead of using the hard coded value `2`. It seems quite arbitrary here.

Reply:
a utility function would be ok ? 
* `isContractSign(contract) => contract.nb_signatures > 0`
* `isContractCountersign(contract) => contract.nb_signatures > 1`

Reply:
Yes + with constants too